connectors/connectors-common-library/src/main/java/io/camunda/connector/common/services/HTTPService.java

```diff
@@ -104,15 +112,39 @@ public <T extends CommonResult> T toHttpJsonResponse(
               }
             });
     connectorResult.setHeaders(headers);
-    try (InputStream content = externalResponse.getContent();
-        Reader reader = new InputStreamReader(content)) {
-      final Object body = gson.fromJson(reader, Object.class);
-      if (body != null) {
-        connectorResult.setBody(body);
+    try (InputStream content = externalResponse.getContent()) {
+      String bodyString = null;
+      if (content != null) {
+        bodyString = new String(content.readAllBytes(), StandardCharsets.UTF_8);
+      }
+
+      if (bodyString != null) {
+        if (isJSONValid(bodyString)) {
+          Object body = gson.fromJson(bodyString, Object.class);
+          connectorResult.setBody(body);
+        } else {
+          try {
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder builder = factory.newDocumentBuilder();
+            Document doc = builder.parse(new InputSource(new StringReader(bodyString)));
+            connectorResult.setBody(doc);
```

Comment:
What does this block do? Parsing XML?
If so, please it's perfect place to leave an explanation comment in code. Also, rename the method, since it's no longer does `toHttpJsonResponse`